fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/predicates/OrPredicate.java

```diff
@@ -243,7 +244,8 @@ public Optional<PredicateMultiMap.PredicateMapping> impliesCandidatePredicate(@N
                 //
                 return Optional.of(PredicateMultiMap.PredicateMapping.orTermMapping(this,
                         new ConstantPredicate(true),
-                        getDefaultCompensatePredicateFunction()));
+                        getDefaultCompensatePredicateFunction(),
```

Comment:
I don't think these are right. Don't you need to use the proper translation map (that gets derived from the underlying  M3 maps)? Same with the order predicate kinds...

Reply:
I set the calculation of the translated predicate to `Optional.empty()`.

I think you're right, however the translation maps are providing means to translate a value between a query result value and a candidate result value, however, for predicates, I think we should only pull up the predicates along the candidate's expressions. In other words, one of them is horizontal, while the other is vertical if that makes sense.

Reply:
I am not sure I follow your comment fully, however, I think it doesn't matter for this PR.